src/main/java/org/folio/entlinks/integration/kafka/AuthorityEventListener.java

```diff
@@ -33,20 +40,29 @@ public class AuthorityEventListener {
   public void handleEvents(List<ConsumerRecord<String, AuthorityDomainEvent>> consumerRecords) {
     log.info("Processing authorities from Kafka events [number of records: {}]", consumerRecords.size());
 
-    var authorityEvents =
-      consumerRecords.stream()
-        .map(consumerRecord -> {
-          var value = consumerRecord.value();
-          value.setId(UUID.fromString(consumerRecord.key()));
-          return value;
-        })
-        .collect(Collectors.groupingBy(AuthorityDomainEvent::getTenant));
+    consumerRecords.stream()
+      .collect(Collectors.groupingBy(consumerRecord ->
+        extractHeaderValue(XOkapiHeaders.USER_ID, consumerRecord.headers())))
+      .forEach(this::handleAuthorityEventsForUser);
```

Comment:
I don't think this grouping will positively affect performance, if not make it worse.
It will be better to handle headers for each consumer record separately.
Or at least move it after tenant grouping.

Reply:
Handling each consumer record separately means no batch processing, I moved it after tenant grouping